From b4876eaeba333192e1ea692aaed787467678cb7d Mon Sep 17 00:00:00 2001
From: rleix <rayx.lei@intel.com>
Date: Mon, 4 May 2020 15:20:28 +0530
Subject: [PATCH] Enable add user button if the dialog canceled before

If the add-user dialog was canceled by clicking outside of dialog,
Then user will not able to create new user anymore since the button
was disabled. Add "onCancel" method to enable the button again.

Change-Id: Ie3200e9d047c4f9542fbca9484d663b47bc7cdc2
Tracked-On: OAM-80786
Signed-off-by: rleix <rayx.lei@intel.com>
---
 .../car/settings/common/ConfirmationDialogFragment.java    | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git src/com/android/car/settings/common/ConfirmationDialogFragment.java src/com/android/car/settings/common/ConfirmationDialogFragment.java
index ee1d54c7..cbe17045 100644
--- src/com/android/car/settings/common/ConfirmationDialogFragment.java
+++ src/com/android/car/settings/common/ConfirmationDialogFragment.java
@@ -35,6 +35,7 @@ import com.android.car.ui.preference.CarUiDialogFragment;
  * action.
  */
 public class ConfirmationDialogFragment extends CarUiDialogFragment {
+        DialogInterface.OnClickListener, DialogInterface.OnCancelListener {
 
     /** Builder to help construct {@link ConfirmationDialogFragment}. */
     public static class Builder {
@@ -263,6 +264,12 @@ public class ConfirmationDialogFragment extends CarUiDialogFragment {
             }
         }
     }
+    @Override
+    public void onCancel(DialogInterface dialog) {
+        if (mRejectListener != null) {
+            mRejectListener.onReject(getArguments().getBundle(ARGUMENTS_KEY));
+        }
+    }
 
     /** Listens to the confirmation action. */
     public interface ConfirmListener {
-- 
2.17.1

